src/passes/publicFunctionSplitter/internalFunctionCallCollector.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  FunctionCall,
+  FunctionCallKind,
+  FunctionDefinition,
+  Identifier,
+  MemberAccess,
+} from 'solc-typed-ast';
+import { AST } from '../../ast/ast';
+import { ASTMapper } from '../../ast/mapper';
+
+export class InternalFunctionCallCollector extends ASTMapper {
+  constructor(public internalFunctionCallSet: Set<FunctionDefinition>) {
+    super();
+  }
+
+  /*
+     This class collects all internal function calls in the contract
+  */
+  visitFunctionCall(node: FunctionCall, ast: AST): void {
+    const functionDefintion = node.vReferencedDeclaration;
+    if (
+      node.kind === FunctionCallKind.FunctionCall &&
+      functionDefintion instanceof FunctionDefinition &&
+      (node.vExpression instanceof MemberAccess || node.vExpression instanceof Identifier)
+    ) {
+      this.internalFunctionCallSet.add(functionDefintion);
```

Comment:
Since this pass is run on all the roots before with a single `internalFunctionCallSet` will it not also collect external calls?

Reply:
Yes, it also collects external calls but since in the moment the pass is done I have no information of the Functions defined in the contract. It would be necessary to add